_gsocblogs/2022/blog_Rucio_NimishBongale.md

```diff
@@ -1,127 +1,325 @@
+
+#### Navigation + Search
+
+The left side bar navigation brings in a new dashboard-like flavour to the WebUI. The Search bar on top has been preserved as is in terms of functionality.
+
+#### Protected Routes
+
+`react-router-dom` has been configured to handle protected routes, and allow only authenticated users to access certain protected routes. In case of an invalid access the user is redirected to the Login page.
+
+#### Global Store
+
+Making use of the React Context API, a global store is available. This implementation creates a context, which could be used to access the global store from anywhere within the application. An updateStore function is provided to publish updates to it. This causes any component subscribing to the store to update itself with the fresh props.
+
+#### Alert + Modal Hooks
+
+Since alerts/modals can be shown from anywhere within the application, special purpose-built hooks were developed, which can be instantiated and called. No additional imports are necessary - the content that these popups display is entirely customizable as well.
+
+#### Responsive
+
+Mockups were built both for the desktop and mobile versions. Care has been taken from Day 1 to build views that were fully responsive.
+
+### Best Development Practices
+
+#### CI/CD
+
+Custom CI/CD was in place since the first week of the project. It not only builds both the React app & Storybook app, but also deploys Storybook to GitHub pages after a successful merge to master.
+
+#### Code Structure
+
+Types, Models, Exceptions & REST Wrapper not only gave birth to the `RucioClient`, but also enabled a cleaner and more readable codebase. The storybook components sit neatly within the `stories` folder, and the React pages sit within `components`.
+
+#### Code Quality
+
+On top of type checking, variable usages are null-safed, object and array destructing is in place wherever necessary and exceptions are handled appropriately.
+
+#### Unit Testing
+
+Using the react testing library, Unit Testing has played a crucial role in building a resilient and bug-free codebase. While there is a lot more ground to cover in terms of coverage, the ability to test at a component level will greatly simplify the overall testing.
+
+#### Accessibility
+
+a11y plays a pivotal role in determining if an application caters to audience of all forms. Using WAVE and other accessibility tools, there has been a solid attempt at improving the a11y score of the website since the day the first page was complete.
+
+### Relevant PRs
+
+- [Code Refactoring](https://github.com/rucio/webui/pull/98)
+
+#98 by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+author%3Animishbongale) was merged 4 days ago
```

Comment:
Please remove all the lines like this showing the status of the PR, just keep the links to the PRs